anndata/_core/raw.py

```diff
@@ -34,7 +34,10 @@ def __init__(
                 self._X = X.get()
             else:
                 self._X = X
-            self._var = _gen_dataframe(var, self.X.shape[1], ["var_names"])
+            n_var = None if self._X is None else self._X.shape[1]
+            self._var = _gen_dataframe(
+                var, ["var_names"], source="X", attr="var", length=n_var
+            )
```

Comment:
`_gen_dataframe` used to ignore `length` before when the first argument was already a `DataFrame`.

I assume passing `self.X.shape[1]` was therefore simply wrong (note that it’s `self._X.shape[1]` now, with an underscore)

Reply:
Accesses like that are often around backed mode where `_X is None` for anndata. I think it should be fine in this case.

Reply:
OK!